src/sentry/api/permissions.py

```diff
@@ -28,27 +29,41 @@
     from sentry.models.organization import Organization
 
 
-class RelayPermission(permissions.BasePermission):
+class RelayPermission(BasePermission):
     def has_permission(self, request: Request, view: object) -> bool:
         return getattr(request, "relay", None) is not None
 
 
-class SystemPermission(permissions.BasePermission):
+class SystemPermission(BasePermission):
     def has_permission(self, request: Request, view: object) -> bool:
         return is_system_auth(request.auth)
 
 
-class NoPermission(permissions.BasePermission):
+class NoPermission(BasePermission):
     def has_permission(self, request: Request, view: object) -> bool:
         return False
 
 
-class SuperuserPermission(permissions.BasePermission):
+class SuperuserPermission(BasePermission):
     def has_permission(self, request: Request, view: object) -> bool:
         return is_active_superuser(request)
 
 
-class ScopedPermission(permissions.BasePermission):
+class StaffPermission(BasePermission):
+    def has_permission(self, request: Request, view: object) -> bool:
+        return is_active_staff(request)
+
+
+class SuperuserOrStaffFeatureFlaggedPermission(BasePermission):
+    def has_permission(self, request: Request, view: object) -> bool:
+        has_staff = features.has("auth:enterprise-staff-cookie", actor=request.user)
+        active_superuser = not has_staff and SuperuserPermission().has_permission(request, view)
+        active_staff = has_staff and StaffPermission().has_permission(request, view)
```

Comment:
```suggestion
        active_superuser = SuperuserPermission().has_permission(request, view)
        active_staff = False
        if has_staff_feature_flag:
            active_staff = StaffPermission().has_permission(request, view)
```
nit: can we also refactor this so it's easier to see the effects of the feature flag? from my understanding we only actually check `StaffPermission().has_permission` if we have the feature flag?

Reply:
on second thought i may be misunderstanding what this is doing, feel free to correct me